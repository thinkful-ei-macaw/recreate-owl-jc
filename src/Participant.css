.participant-img {
  height: 50px;
  width: 50px;
  width: 10%;
}

.participant-card {
  height: 100px;
  width: 400px;
  display: flex;
  flex-direction: row;
}

.stage-status {
  width: 20%;
}

.name-paragraph {
  width: 70%;
}