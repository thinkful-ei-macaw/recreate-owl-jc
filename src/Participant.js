import React from 'react'

const Participant = (props) => {

return (
<div>
  <img className="participant participant-img" alt="Avatar" src={props.avatar}/>
  <p>{props.name}</p>
  {props.onStage ? <p>on stage</p> : null}
</div>
);
}

export default Participant;