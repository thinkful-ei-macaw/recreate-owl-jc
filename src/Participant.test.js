import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participant from './Participant';


describe('Participant Component Tests', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Participant name="Koren Templeton" avatar={'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'} inSession={true} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the Participant with correct info', () => {
    const tree = renderer
    .create(<Participant name="Koren Templeton" avatar={'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'} inSession={true} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
