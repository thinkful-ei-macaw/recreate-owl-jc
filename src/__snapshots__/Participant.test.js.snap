// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant Component Tests renders the Participant with correct info 1`] = `
<div>
  <img
    alt="Avatar"
    className="participant participant-img"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <p>
    Koren Templeton
  </p>
</div>
`;
